homeassistant/components/ezviz/camera.py

```diff
@@ -0,0 +1,236 @@
+            camera_rtsp_stream = f"rtsp://{camera_username}:{camera_password}@{camera['local_ip']}:{local_rtsp_port}"
+            _LOGGER.debug(
+                "Camera %s source stream: %s", camera["serial"], camera_rtsp_stream
+            )
+
+        else:
+            _LOGGER.info(
+                "I found a camera (%s) but it is not configured. Please configure it if you wish to see the appropriate stream. Conf cameras: %s",
+                camera_serial,
+                conf_cameras,
+            )
+
+        camera["username"] = camera_username
+        camera["password"] = camera_password
+        camera["rtsp_stream"] = camera_rtsp_stream
+
+        camera["ezviz_camera"] = EzvizCamera(ezviz_client, camera_serial)
+
+        camera_entities.append(HassEzvizCamera(**camera))
+
+    add_entities(camera_entities)
+
+
+class HassEzvizCamera(Camera):
+    """An implementation of a Foscam IP camera."""
+
+    def __init__(self, **data):
+        """Initialize an Ezviz camera."""
+        super().__init__()
+
+        self._username = data["username"]
+        self._password = data["password"]
+        self._rtsp_stream = data["rtsp_stream"]
+
+        self._ezviz_camera = data["ezviz_camera"]
+        self._serial = data["serial"]
+        self._name = data["name"]
+        self._status = data["status"]
+        self._privacy = data["privacy"]
+        self._audio = data["audio"]
+        self._ir_led = data["ir_led"]
+        self._state_led = data["state_led"]
+        self._follow_move = data["follow_move"]
+        self._alarm_notify = data["alarm_notify"]
+        self._alarm_sound_mod = data["alarm_sound_mod"]
+        self._encrypted = data["encrypted"]
+        self._local_ip = data["local_ip"]
+        self._detection_sensibility = data["detection_sensibility"]
+        self._device_sub_category = data["device_sub_category"]
+        self._local_rtsp_port = data["local_rtsp_port"]
+
+        self._ffmpeg = None
+
+    def update(self):
+        """Update the camera states."""
+
+        data = self._ezviz_camera.status()
+
+        self._name = data["name"]
+        self._status = data["status"]
+        self._privacy = data["privacy"]
+        self._audio = data["audio"]
+        self._ir_led = data["ir_led"]
+        self._state_led = data["state_led"]
+        self._follow_move = data["follow_move"]
+        self._alarm_notify = data["alarm_notify"]
+        self._alarm_sound_mod = data["alarm_sound_mod"]
+        self._encrypted = data["encrypted"]
+        self._local_ip = data["local_ip"]
+        self._detection_sensibility = data["detection_sensibility"]
+        self._device_sub_category = data["device_sub_category"]
+        self._local_rtsp_port = data["local_rtsp_port"]
+
+    async def async_added_to_hass(self):
+        """Subscribe to ffmpeg and add camera to list."""
+        self._ffmpeg = self.hass.data[DATA_FFMPEG]
+        entities = self.hass.data.setdefault(EZVIZ_DATA, {}).setdefault(ENTITIES, [])
```

Comment:
This is never used. Storing the entities doesn't seem needed.

Reply:
yep, stale code from the new version. sorry